test/cmd/lfstest-gitserver.go

```diff
@@ -228,19 +228,35 @@ func lfsBatchHandler(w http.ResponseWriter, r *http.Request, repo string) {
 	res := []lfsObject{}
 	testingChunked := testingChunkedTransferEncoding(r)
 	for _, obj := range objs.Objects {
-		o := lfsObject{
-			Oid:  obj.Oid,
-			Size: obj.Size,
-			Actions: map[string]lfsLink{
-				"upload": lfsLink{
-					Href:   lfsUrl(repo, obj.Oid),
-					Header: map[string]string{},
+		_, ok := largeObjects.Get(repo, obj.Oid)
+		var o lfsObject
+		if ok {
+			o = lfsObject{
+				Oid:  obj.Oid,
+				Size: obj.Size,
+				Actions: map[string]lfsLink{
+					"download": lfsLink{
+						Href: lfsUrl(repo, obj.Oid),
+					},
 				},
-			},
-		}
-
-		if testingChunked {
-			o.Actions["upload"].Header["Transfer-Encoding"] = "chunked"
+			}
+			if testingChunked {
+				o.Actions["download"].Header["Transfer-Encoding"] = "chunked"
+			}
+		} else {
+			o = lfsObject{
+				Oid:  obj.Oid,
+				Size: obj.Size,
+				Actions: map[string]lfsLink{
+					"upload": lfsLink{
+						Href:   lfsUrl(repo, obj.Oid),
+						Header: map[string]string{},
+					},
+				},
+			}
+			if testingChunked {
+				o.Actions["upload"].Header["Transfer-Encoding"] = "chunked"
+			}
```

Comment:
This duplication seems unnecessary. Can we get away with changing the action key from "upload" to "download" based on the bool return value from `largeObjects.Get()`?